### PR TITLE
SPITFIRE - ignitions, area burn, intensity, scorch height

### PR DESCRIPTION
**Pull request by jkshuman**
### Description:
This set of changes addresses fire behavior and effects updates to equations. A cloud to ground scaler on lightning ignitions is introduced per Latham and Williams (2001) (addresses #593). Ignitions must be greater than 0 to begin calculations of fire behavior (addresses issue #589).  Area burn and fire intensity subroutines are combined to allow for inclusion of fraction of patch burnt in calculation of fire intensity (FI) per Thonicke et al 2010. Ignition probability (FDI), fire duration (FD), area burnt (AB) and frac_burnt now calculated prior to FI to include this dependancy. Scorch height is updated to be at the cohort level and scaled by PFT trait "alpha_SH" using the patch level fire intensity (FI) per Van Wagner (1973) (addresses issue #477 and #476). This update maintains the fire threshold of 50 kW/m for fire events per Pyne (1986). 

This addresses issues #593 #589 #477 and #476

### Collaborators:
Discussions with @adrifoster @pollybuotte @rgknox @rosiealice @glemieux 

### Expectation of Answer Changes:
Updates will result in less ignitions, less area burn, and lower fire intensity compared to current master. Attached figure with difference between branch with update to ignitions only and branch using all ignitions for burned area (BA) and GPP after 4 years of simulation.
Ignitions_CG_ign-all_ign_yr4

### Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the in-code documentation .AND. (the [technical note](https://github.com/NGEET/fates-docs) .OR. the wiki) accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/NGEET/fates/blob/master/CONTRIBUTING.md) document.
- [ ] FATES PASS/FAIL regression tests were run
- [x] If answers were expected to change, evaluation was performed and provided

### Test Results:
TBD

CTSM (or) E3SM (specify which) test hash-tag:

CTSM (or) E3SM (specify which) baseline hash-tag:

FATES baseline hash-tag:

Test Output:

### CODE REVIEW

File: biogeochem/EDCohortDynamicsMod.F90 [hunk 1]
```diff
@@ -534,6 +534,7 @@ subroutine nan_cohort(cc_p)
     currentCohort%ddbhdt             = nan ! time derivative of dbh 
 
     ! FIRE
+    currentCohort%sh                    = nan ! scorch height affecting crown (m)
```
Comment (rgknox) [inline biogeochem/EDCohortDynamicsMod.F90]: could we make this variable a little more verbose? When I see sh without context, my mind goes to specific humidity.

Comment (jkshuman) [reply biogeochem/EDCohortDynamicsMod.F90]: yep. will update now.

**Comment by jkshuman** [comment]
@rgknox thank you for the thorough review of these changes, and glad to hear they pass muster.

I agree that the commented out portion of the ellipse fire shape and size feedback isn't great for readability. I do plan on improving this section of the code in the near future, so I am inclined to leave it here for the moment.

The fire_threshold does have actual meaning, and the active line uses a value from the literature (Pyne 1986) that identifies fires that go out on their own (too small to spread). The fire-threshold set close to zero, was commented out, so I removed that. Tracking all fires with energy close to zero needs more evaluation of the impact on area burnt and other values (veg mortality, etc). I will be cleaning that section of the code in the near future. (This will likely overlap with the modification of the ellipse fire shape and size portion of the code.)

**Comment by glemieux** [comment]
@jkshuman some of the current standard test suite call the `SCORCH_HEIGHT history variable`:
https://github.com/ESCOMP/CTSM/blob/b350e06e4615c124d3ef1d8f3f9831a1f8f3f402/cime_config/testdefs/testmods_dirs/clm/Fates/user_nl_clm#L1-L7
As such, the test runs are crashing trying to find it.  Do you want to keep `SCORCH_HEIGHT` available for future updates, or should I create a PR to update the test configuration files to remove it?

**Comment by jkshuman** [comment]
@glemieux with the switch of scorch_ht to the cohort (PFT) level, we don't need this (scorch height) for future updates. You can get scorch height from the fire intensity. So, yes, my recommendation is to remove it from the test configuration files.

**Comment by glemieux** [comment]
> @glemieux with the switch of scorch_ht to the cohort (PFT) level, we don't need this (scorch height) for future updates. You can get scorch height from the fire intensity. So, yes, my recommendation is to remove it from the test configuration files.

@rgknox do we want to remove the commented lines out of FatesHistoryInterfaceMod.F90 given Jackie's recommendation?

**Comment by jkshuman** [comment]
@rgknox @glemieux oops. yeah, meant to check in with folks on that. The switch to cohort level made the previous patch level use invalid. Got hasty in my implementation. Let me know if you want me to do that, or if you want to open up this decision to more input. I can add it back in with cohort level summary in a future PR, but it should not be an active output.

**Comment by glemieux** [comment]
> I can add it back in with cohort level summary in a future PR, but it should not be an active output.

Sorry for the newbie question @jkshuman : how would a cohort level summary be implemented?

**Comment by jkshuman** [comment]
@glemieux not a bad question. I believe this was why I left it out. I ran out of time to investigate the answer to that question for myself in the mad dash prior to AGU...

**Comment by jkshuman** [comment]
@glemieux I also don't think it is absolutely necessary, as you can get this from fire intensity. But, I should have cleaned up the history file interface. I think I left it there so we could have this discussion about it being removed...I wanted to revisit when I wasn't in a panic for time.

**Comment by ckoven** [comment]
@jkshuman lets just delete it the commented history code.  we can always pull it out of old commits if someone finds they have a need for it.

**Comment by pollybuotte** [comment]
For what it's worth @jkshuman @glemieux , I'll want to look at scorch height as we compare the active and passive crown fire implementations. If it's not a history variable, is there a place to hold the calculation from fire intensity that will be readily available (i.e. so we don't have to ask you, Jackie!)?

**Comment by jkshuman** [comment]
@pollybuotte that calculation is in the spitfire code
https://github.com/jkshuman/fates/blob/e39aa5f27f2d88b8bae6ecf1b906b5f4fdd31496/fire/SFMainMod.F90#L839-L840
@ckoven sure thing. I will can remove those lines.

**Comment by ckoven** [comment]
@pollybuotte @jkshuman it looks like this is really fundamentally a patch x PFT property rather  than a cohort property.  in principle we have the machinery to output such variables; e.g. see here: https://github.com/NGEET/fates/blob/e39aa5f27f2d88b8bae6ecf1b906b5f4fdd31496/main/FatesHistoryInterfaceMod.F90#L2201

in the interest of limiting PR scope-creep, maybe let's add that to a near-future PR?

**Comment by jkshuman** [comment]
@ckoven you are correct, and sounds good to me.

**Comment by glemieux** [comment]
Chatting with @ckoven prior to the all hands discussion today, we decided that since the three test definitions that call `SCORCH_HEIGHT` also call another fire variable (i.e. `FIRE_INTENSITY`), we can likely be safe in striking `SCORCH_HEIGHT` from the test definitions even though we may reuse that variable name in the future PR.

**Comment by jkshuman** [comment]
> @pollybuotte @jkshuman it looks like this is really fundamentally a patch x PFT property rather than a cohort property. in principle we have the machinery to output such variables; e.g. see here:
> 
> https://github.com/NGEET/fates/blob/e39aa5f27f2d88b8bae6ecf1b906b5f4fdd31496/main/FatesHistoryInterfaceMod.F90#L2201
> 
> in the interest of limiting PR scope-creep, maybe let's add that to a near-future PR?

@ckoven if this is not too time consuming per your example, I can add this into the history interface with this PR. and then we would not need to update the testing definitions @glemieux ?
though, I have somehow broken my Hobart git environment, and cannot create cases at the moment, so I am in a holding pattern until I can fix that.

**Comment by glemieux** [comment]
@jkshuman that would make things slightly less complex from my standpoint, since I wouldn't have to coordinate a PR with the HLM.  BTW, I did run a test on my local branch with 'SCORCH_HEIGHT` removed from the test definitions and everything passed as expected (only DIFF being in `FIRE_INTENSITY`).

**Comment by ckoven** [comment]
@jkshuman sounds good.  actually if its ok with you, I can just write that history code myself and submit a PR to your branch?  I'm keen to redo an old set of gridded california simulations with your new fire code and so can do this first before I kick off runs...

**Comment by jkshuman** [comment]
@ckoven that would be great, as I need to fix my git environment. let me know how things look in CA. weird cime error.

**Comment by glemieux** [comment]
Pulled in the updates and re-testing.

**Comment by glemieux** [comment]
Test Results:

All expected PASS.  FAILs only on FatesHydro RUN and test cases with DIFFs for `FIRE_INTENSITY`.

CTSM (or) E3SM (specify which) test hash-tag: 5ad8a8d2

CTSM (or) E3SM (specify which) baseline hash-tag: 097f4239

FATES baseline hash-tag: 17aaa879

Test Output:
```
/glade/u/home/glemieux/scratch/clmed-tests/pr594-fire-threshold.fates.cheyenne.intel.C5ad8a8d2-F931824c8
```

**Pull request merged**